src/mds/Mutation.cc

```diff
@@ -361,6 +363,12 @@ bool MDRequestImpl::is_queued_for_replay() const
   return client_request ? client_request->is_queued_for_replay() : false;
 }
 
+bool MDRequestImpl::is_batch_op()
+{
+  return client_request->get_op() == CEPH_MDS_OP_LOOKUP
+    || client_request->get_op() == CEPH_MDS_OP_GETATTR;
+}
```

Comment:
(client_request == CEPH_MDS_OP_LOOKUP &&
client_request->get_filepath().depth() == 1) ||
(client_request->get_op() == CEPH_MDS_OP_GETATTR &&
client_request->get_filepath().depth() == 0)

Reply:
@ukernel I don't quite follow you, why do we have to restrict the filepath length? Thanks:-)

Reply:
To handle requests in batch.  All requests should require same set of locks.  longer filepath require extra locks.

Reply:
@ukernel I got it:-)